UIKitLesson1/PlayerVC.swift

```diff
@@ -0,0 +1,214 @@
+            durationLabel.bottomAnchor.constraint(equalTo: currentTimeSlider.topAnchor, constant: 5),
+            
+            previosTrackButton.topAnchor.constraint(equalTo: currentTimeSlider.bottomAnchor, constant: 15),
+            previosTrackButton.leftAnchor.constraint(equalTo: view.leftAnchor, constant: 50),
+            
+            playPauseButton.topAnchor.constraint(equalTo: currentTimeSlider.bottomAnchor, constant: 15),
+            playPauseButton.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            
+            nextTrackButton.topAnchor.constraint(equalTo: currentTimeSlider.bottomAnchor, constant: 15),
+            nextTrackButton.rightAnchor.constraint(equalTo: playPauseButton.rightAnchor, constant: 85)
+        ])
+    }
+    
+    private func setupPlayer() {
+        let song = songs[position]
+        let audioPath = Bundle.main.path(forResource: song.trackName, ofType: "mp3")
+        
+        do {
+            try AVAudioSession.sharedInstance().setMode(.default)
+            try AVAudioSession.sharedInstance().setActive(true, options: .notifyOthersOnDeactivation)
+            
+            guard let audioPath = audioPath else { return }
+            player = try AVAudioPlayer(contentsOf: URL(string: audioPath)!)
+            
+            player.play()
+            
+        } catch {
+            print("error")
+        }
+        
+        setButtons()
+        
+        trackImageView.image  = UIImage(named: song.imageName)
+        albumTitleLabel.text  = song.albumName
+        trackTitleLabel.text  = song.name
+        authorTitleLabel.text = song.artistName
+        currentTimeLabel.text = "00:00"
+        durationLabel.text    = "--:--"
+        [albumTitleLabel, trackTitleLabel, authorTitleLabel].forEach{ $0.font = .boldSystemFont(ofSize: 20)}
+    }
+    
+    private func setButtons() {
+        hideButton.setImage(.init(systemName: "chevron.down"), for: .normal)
+        hideButton.addTarget(self, action: #selector(dragDownButtonTapped), for: .touchUpInside)
+        
+        shareButton.setImage(UIImage(systemName: "square.and.arrow.up"), for: .normal)
+        shareButton.addTarget(self, action: #selector(shareButtonTapped), for: .touchUpInside)
+        
+        previosTrackButton.setImage(UIImage(named: "pre"), for: .normal)
+        previosTrackButton.addTarget(self, action: #selector(previosTrack), for: .touchUpInside)
+        
+        playPauseButton.setImage(UIImage(named: "pause"), for: .normal)
+        playPauseButton.addTarget(self, action: #selector(playPouseAction), for: .touchUpInside)
+        
+        nextTrackButton.setImage(UIImage(named: "next"), for: .normal)
+        nextTrackButton.addTarget(self, action: #selector(nextTrack), for: .touchUpInside)
+        
+        currentTimeSlider.addTarget(self, action: #selector(playbackSliderValueChanged), for: .valueChanged)
+    }
+    
+    @objc func playbackSliderValueChanged(_ playbackSlider: UISlider) {
```

Comment:
```suggestion
    @objc 
    private func playbackSliderValueChanged(_ playbackSlider: UISlider) {
```

а почему экшены не приватные?)